modules/mail/src/main/java/org/apache/axis2/transport/mail/MailTransportSender.java

```
@@ -257,14 +257,34 @@ private String sendMail(MailOutTransportInfo outInfo, MessageContext msgContext)
         }
 
         WSMimeMessage message = null;
-        if (outInfo.getFromAddress() != null) {
+        Map trpHeaders = (Map) msgContext.getProperty(MessageContext.TRANSPORT_HEADERS);
+        if (trpHeaders != null
+                && trpHeaders.containsKey(MailConstants.MAIL_HEADER_FROM)
+                && trpHeaders.containsKey(MailConstants.MAIL_SMTP_PASSWORD)
+                && trpHeaders.containsKey(MailConstants.MAIL_SMTP_USERNAME)) {
+            Properties prop = (Properties) session.getProperties().clone();
+            final String from = (String) trpHeaders.get(MailConstants.MAIL_HEADER_FROM);
+            final String password = (String) trpHeaders.get(MailConstants.MAIL_SMTP_PASSWORD);
+            final String userName = (String) trpHeaders.get(MailConstants.MAIL_SMTP_USERNAME);
+            prop.setProperty(MailConstants.MAIL_HEADER_FROM, from);
+            prop.setProperty(MailConstants.MAIL_SMTP_USERNAME, userName);
+            prop.setProperty(MailConstants.MAIL_SMTP_PASSWORD, password);
+
+            Session sessionCurrent = Session.getInstance(prop, new Authenticator() {
+                public PasswordAuthentication getPasswordAuthentication() {
+
+                    return new PasswordAuthentication(userName, password);
+                }
+            });
+
+            message = new WSMimeMessage(sessionCurrent, (String) trpHeaders.get(MailConstants.MAIL_HEADER_FROM));
+        } else if (outInfo.getFromAddress() != null) {
             message = new WSMimeMessage(session, outInfo.getFromAddress().getAddress());
         } else {
             message = new WSMimeMessage(session, "");
```

Is this a valid use-case? Should we throw an error if input are not provided?